src/core/ignore.c

```diff
@@ -326,26 +322,23 @@ static void ignore_remove_config(IGNORE_REC *rec)
 
 static void ignore_init_rec(IGNORE_REC *rec)
 {
-#ifdef HAVE_REGEX_H
-	char *errbuf;
-	int errcode, errbuf_len;
+	if (rec->regexp_compiled) {
+		g_regex_unref(rec->preg);
+		rec->regexp_compiled = FALSE;
+	}
 
-	if (rec->regexp_compiled) regfree(&rec->preg);
-	rec->regexp_compiled = FALSE;
 	if (rec->regexp && rec->pattern != NULL) {
-		errcode = regcomp(&rec->preg, rec->pattern,
-				REG_EXTENDED|REG_ICASE|REG_NOSUB);
-		if (errcode != 0) {
-			errbuf_len = regerror(errcode, &rec->preg, 0, 0);
-			errbuf = g_malloc(errbuf_len);
-			regerror(errcode, &rec->preg, errbuf, errbuf_len);
-			g_warning("Failed to compile regexp '%s': %s", rec->pattern, errbuf);
-			g_free(errbuf);
+		GError *re_error;
+
+		rec->preg = g_regex_new(rec->pattern, G_REGEX_CASELESS | G_REGEX_EXTENDED, 0, &re_error);
```

Comment:
My bad, here's an excerpt from `pcreposix(3)` for more informations.

    I have implemented only those POSIX option bits that can be
    reasonably mapped to PCRE native options. In addition, the option
    REG_EXTENDED is defined with the value zero. This has no effect, but
    since programs that are written to the POSIX interface often use it,
    this makes it easier to slot in PCRE as a replacement library. Other
    POSIX options are not even defined.